content/docs/concepts/components/builder.md

```diff
@@ -1,7 +1,7 @@
 +++
 title="Builder"
 weight=1
-summary="A builder is an image that bundles all the bits and information on how to build your apps, such as buildpacks, an implementation of the lifecycle, and a build-time environment that platforms may use when executing the lifecycle."
+summary="A builder is an image that bundles all the bits and information on how to build your app. It contains buildpacks, an implementation of the lifecycle program, and a build-time environment that platforms may use when executing the lifecycle."
```

Comment:
A builder can most certainly build multiple applications. Not at once of course but the same builder can be used to build java apps, node apps, ruby apps, etc. For example, you can see all the language supports on the [Google builder](https://github.com/GoogleCloudPlatform/buildpacks/blob/main/builders/gcp/base/builder.toml#L103
).
 
```suggestion
summary="A builder is an image that bundles all the bits and information on how to build your apps. It contains buildpacks, an implementation of the lifecycle program, and a build-time environment that platforms may use when executing the lifecycle."
```

Reply:
How about adding it with the separate sentence "The same builder can build apps in different languages. When analysing code, it automatically selects appropriate buildpack for the target language." ?

Reply:
I'm wary about using singular _app_ because I would hate for the initial perception to be that there's a one-to-one relationship between app and builder. What are your concerns with a plural?

In regards to the additional blurb about languages I would say:

1. I think it would be great to add this information but I wouldn't put it in the summary. I would add another paragraph talking about the one-to-many potentiality.
2. The idea of a single builder supporting multiple languages is just an example and when added it should be noted as such. Another example could be that you can simple have a javascript builder (if you are a strict javascript shop) that understands various development stacks (ie. node with yarn vs node with npm, etc).


Maybe something like (a _very rough_ draft):

> Builders are typically not created specifically for one application. Instead, they are composed of multiple buildpacks that when the code is analyzed the proper buildpacks are operate against it.
>
> For example, the same builder can build apps in different languages. When analysing code, it would automatically select the appropriate buildpacks for the target language.
>

Reply:
My concerns with plural as a DevOps is that the relationship between a builder and produced artifact is one-to-one. If I need to get multiple artifacts (back, front and worker), I need to run the builder multiple times with multiple targets. Compare this to Kubernetes - I run it once and it executes multiple artifacts. That's how one-to-many relationship works for me.

Reply:
> Another example could be that you can simple have a javascript builder (if you are a strict javascript shop) that understands various development stacks (ie. node with yarn vs node with npm, etc).

I thought that builder can't understand anything. It is the role of buildpack to analyse the project and tell if the buildpack can build it, no?

> Builders are typically not created specifically for one application. Instead, they are composed of multiple buildpacks that when the code is analyzed the proper buildpacks are operate against it.

This leaves too many open questions. Why builders are composed of buildpacks? Who and when analyses the code? If buildpacks operate against code on their own, why builder is needed at all?